### PR TITLE
Milestone 2: Components and Redux store

### PR DESCRIPTION
**Pull request by shubhsk88**
This PR addresses the implementation of the bookstore file structure and initialization of redux in the web app

### CODE REVIEW

File: src/components/BooksList/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
+import Book from '../Book';
+
+function BooksList({ books }) {
+  return (
+    <table>
+      <thead>
+        <tr>
+          <th>ID</th>
```
Comment (bellom) [inline src/components/BooksList/index.js]: opps, we need to rearrange some file.
React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory (presentational components) you have components that are not connected to the Redux store, and on your containers directory you have components that are connected to the Redux store. https://redux.js.org/basics/usage-with-react#presentational-and-container-components 

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

File: src/index.js [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
 
+const store = createStore(rootReducer);
 ReactDOM.render(
-  <React.StrictMode>
+  <Provider store={store}>
     <App />
-  </React.StrictMode>,
-  document.getElementById('root')
+  </Provider>,
+  document.getElementById('root'),
```
Comment (bellom) [inline src/index.js]: we need to move the book object created in `reducer/book.js` in here.

image

File: src/reducers/books.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { CREATE_BOOK } from '../actions';
+
+const initialState = [
+  { id: 1, title: 'Scream', category: 'Horror' },
+  { id: 2, title: 'Bullets', category: 'Action' },
+];
```
Comment (bellom) [inline src/reducers/books.js]: let's move the book object to `src/index.js`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/components/BooksList/index.js (hunk 1 above)
Comment (shubhsk88) [reply src/components/BooksList/index.js]: This article is too old and the writer wrote not to follow this in current practice

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi team :hand: 

As it was pointed out by @bellom, the approach we need to use in this app is the one specified in the instructions, the **Presentational and Container Components**. The page provided by @bellom it's actually not old. Those are the official guides and they are regularly updated. But I understand what you meant. Today we have other approaches and more recently created designs pattern. But it does not mean that the Presentational and Container components is outdated, it's still widely used on React-Redux applications along with the new react hooks. Each approach has it's pros and cons. But since this is the most common approach when your app uses Redux, Microverse instructions require its use. I'm really sorry about this, I understand your frustration and I know you want to use a more recently created approach. But I'm not allowed to approve the PR that does not follow that approach. I made a mistake to not notice that when I reviewed milestone 4. But we need to apply this change on your application. Again, I'm really sorry for not noticing that before (when I reviewed milestone 4) team :disappointed: 

Please kindly submit for code review again after making these changes.
Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: 
Hi team :hand: 
Thanks for implementing the requested changes.

Happy Coding :computer: 
**Sergio Torres**

**Pull request closed without merging**